channel/handlers.go

```diff
@@ -668,9 +670,47 @@ func validateSkillWithJob(jobID int16, baseSkillID int32) bool {
 	return true
 }
 
-func (server Server) playerUsePortal(conn mnet.Client, reader mpacket.Reader) {
+func (server Server) playerEnterCashShop(conn mnet.Client, reader mpacket.Reader) {
 	plr, err := server.players.getFromConn(conn)
+	if err != nil {
+		return
+	}
+
+	if plr.summons != nil {
```

Comment:
This can probably be refactored into a function called something along the lines of `plr.hideSummons()` and it probably makes sense to have a `plr.showSummons` as I imagine these would be useful elsewhere as well